ansi/parser.go

```diff
@@ -106,37 +106,38 @@ func (p *Parser) Advance(dispatcher ParserDispatcher, b byte, more bool) parser.
 }
 
 func (p *Parser) collectRune(b byte) {
-	if p.RuneLen < utf8.UTFMax {
-		p.RuneBuf[p.RuneLen] = b
-		p.RuneLen++
+	if p.ParamsLen < utf8.UTFMax {
+		shift := p.ParamsLen * 8
+		p.Cmd &^= 0xff << shift
+		p.Cmd |= int(b) << shift
+		p.ParamsLen++
 	}
 }
```

Comment:
I would suggest this for the readability

```suggestion
	if p.ParamsLen >= utf8.UTFMax {
		return
 	}
 	shift := p.ParamsLen * 8
	p.Cmd &^= 0xff << shift
	p.Cmd |= int(b) << shift
	p.ParamsLen++
 }
```